### PR TITLE
Graph12

### PR DESCRIPTION
**Pull request by Shirai92**

### CODE REVIEW

File: src/Graph12/Graph12.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { ProgressCircle } from "../components/ProgressCircle";
+const data = [
```
Comment (Zorkann) [inline src/Graph12/Graph12.tsx:2]: Please use uppercase for static values :)

File: src/Graph12/Graph12.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+		count: 521,
+	},
+	{
+		label: "CUPLA",
+
+		count: 221,
+	},
+];
+const maxData = 1000;
```
Comment (Zorkann) [inline src/Graph12/Graph12.tsx:24]: Please use uppercase for static values :)

File: src/Graph12/Graph12.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	return (
+		<div className="mt-2">
+			<div className="flex items-center gap-3 w-[250px] ml-3">
+				<div className="bg-blue-500 h-3 w-3 rounded-full block"></div>
+				<span className="uppercase text-[0.5rem]">
+					Lorem Ipsum is simply dummy lorem dummy lorem
+				</span>
+			</div>
+			<div className="flex">
+				{percentages.map((percent, index) => (
```
Comment (Zorkann) [inline src/Graph12/Graph12.tsx:40]: I would use data here and calculate % value directly inside ProgressCircle. Less props You need to pass down the better

File: src/Graph12/Graph12.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+					<ProgressCircle
+						key={index}
```
Comment (Zorkann) [inline src/Graph12/Graph12.tsx:42]: Then you will be able to use label as key, it will be better

File: src/Graph12/Graph12.tsx (hunk 2 above)
Comment (Zorkann) [reply src/Graph12/Graph12.tsx:24]: You can define this value directly inside <ProgressCircle /> or do something like <ProgressCircle max={MAX_DATA} />

File: src/Graph12/Graph12.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+const maxData = 1000;
+
+const percentages = data.map((item) =>
+	Math.round((item.count / maxData) * 100)
+);
+
+export function Graph12() {
+	return (
+		<div className="mt-2">
+			<div className="flex items-center gap-3 w-[250px] ml-3">
```
Comment (Zorkann) [inline src/Graph12/Graph12.tsx:33]: I would remove this ml-3, it doesn't have to be 1:1 as it is on designs

File: src/components/ProgressCircle/ProgressCircle.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+const TRANSPARENT_VARIANTS = {
+	1: {
+		opacity: "0.5",
+	},
+	2: {
+		opacity: "1",
+	},
+};
+
+const COLOR_VARIANTS = {
```
Comment (Zorkann) [inline src/components/ProgressCircle/ProgressCircle.tsx:18]: Please just create 4 variants and add opacity for them if you need, but personally i would not mess up with opacity, i would rather use the same blue color or add another one, slightly different

File: src/components/ProgressCircle/ProgressCircle.tsx [hunk 7]
```diff
@@ -0,0 +1,101 @@
+type ProgressCircleProps = {
+	percent: number;
+	value: number;
+	colorVariant: keyof typeof COLOR_VARIANTS;
+	transparentVariant: keyof typeof TRANSPARENT_VARIANTS;
+	label: string;
+};
+
+const TRANSPARENT_VARIANTS = {
```
Comment (Zorkann) [inline src/components/ProgressCircle/ProgressCircle.tsx:9]: To be removed

File: src/components/ProgressCircle/ProgressCircle.tsx (hunk 6 above)
Comment (Zorkann) [reply src/components/ProgressCircle/ProgressCircle.tsx:18]: I also changed a little how variants are defined, i think we should not use object, just simple array should be enough in our case:
https://github.com/Zorkann/future-dashboard/blob/c6da9c91bab29c97c58362d3e7d939a5fd391cd9/src/components/DottedCircleBlock/DottedCircleBlock.tsx

File: src/components/ProgressCircle/ProgressCircle.tsx [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+};
+
+export function ProgressCircle({
+	percent,
+	value,
+	colorVariant,
+	transparentVariant,
+	label,
+}: ProgressCircleProps) {
+	const circleRadius = 60;
```
Comment (Zorkann) [inline src/components/ProgressCircle/ProgressCircle.tsx:37]: All of static values which does not require props for calculation should be outside of component and uppercase

File: src/Graph12/Graph12.tsx (hunk 1 above)
Comment (Shirai92) [reply src/Graph12/Graph12.tsx:2]: done

File: src/Graph12/Graph12.tsx (hunk 2 above)
Comment (Shirai92) [reply src/Graph12/Graph12.tsx:24]: done

File: src/Graph12/Graph12.tsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+	},
+	{
+		label: "CUPLA",
+
+		count: 221,
+	},
+];
+const maxData = 1000;
+
+const percentages = data.map((item) =>
```
[the remark this answers is not in the document]
Comment (Shirai92) [reply src/Graph12/Graph12.tsx:26]: done

File: src/Graph12/Graph12.tsx (hunk 3 above)
Comment (Shirai92) [reply src/Graph12/Graph12.tsx:40]: done

File: src/Graph12/Graph12.tsx (hunk 4 above)
Comment (Shirai92) [reply src/Graph12/Graph12.tsx:42]: done

File: src/Graph12/Graph12.tsx (hunk 5 above)
Comment (Shirai92) [reply src/Graph12/Graph12.tsx:33]: done

File: src/components/ProgressCircle/ProgressCircle.tsx (hunk 6 above)
Comment (Shirai92) [reply src/components/ProgressCircle/ProgressCircle.tsx:18]: done

File: src/components/ProgressCircle/ProgressCircle.tsx (hunk 7 above)
Comment (Shirai92) [reply src/components/ProgressCircle/ProgressCircle.tsx:9]: done

File: src/components/ProgressCircle/ProgressCircle.tsx (hunk 8 above)
Comment (Shirai92) [reply src/components/ProgressCircle/ProgressCircle.tsx:37]: done

File: src/Graph12/Graph12.tsx [hunk 10]
```diff
@@ -0,0 +1,48 @@
+import { ProgressCircle } from "../components/ProgressCircle";
+const DATA = [
+	{
+		label: "MAGNA",
+
```
Comment (Zorkann) [inline src/Graph12/Graph12.tsx:5]: Please format this data ( remove white space between label and value )

**Pull request merged**